### PR TITLE
Port basic command_buffer tests from the GLSL branch

### PR DESCRIPTION
**Pull request by Kangz**

### CODE REVIEW

File: src/webgpu/api/operation/command_buffer/compute/basic.spec.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  });
+  const bg = t.device.createBindGroup({
+    entries: [
+      { binding: 0, resource: { buffer: src, offset: 0, size: 4 } },
+      { binding: 1, resource: { buffer: dst, offset: 0, size: 4 } },
+    ],
+    layout: bgl,
+  });
+
+  const pl = t.device.createPipelineLayout({ bindGroupLayouts: [bgl] });
```
Comment (austinEng) [inline src/webgpu/api/operation/command_buffer/compute/basic.spec.ts:40]: not sure if we want to rely on it in these tests, but the implicit pipeline layout can be used here

File: src/webgpu/api/operation/command_buffer/render/rendering.spec.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  });
+
+  const colorAttachment = t.device.createTexture({
+    format: 'rgba8unorm',
+    size: { width: 1, height: 1, depth: 1 },
+    usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.OUTPUT_ATTACHMENT,
+  });
+  const colorAttachmentView = colorAttachment.createView();
+
+  const pl = t.device.createPipelineLayout({ bindGroupLayouts: [] });
```
Comment (austinEng) [inline src/webgpu/api/operation/command_buffer/render/rendering.spec.ts:21]: ditto

File: src/webgpu/api/operation/command_buffer/compute/basic.spec.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      entryPoint: 'main',
+    },
+    layout: pl,
+  });
+
+  const encoder = t.device.createCommandEncoder();
+  const pass = encoder.beginComputePass();
+  pass.setPipeline(pipeline);
+  pass.setBindGroup(0, bg);
+  pass.dispatch(1, 1, 1);
```
Comment (austinEng) [inline src/webgpu/api/operation/command_buffer/compute/basic.spec.ts:67]: could be .dispatch(1)

File: src/webgpu/api/operation/command_buffer/render/rendering.spec.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+          `,
+      }),
+      entryPoint: 'main',
+    },
+    layout: pl,
+    primitiveTopology: 'triangle-list',
+    rasterizationState: {
+      frontFace: 'ccw',
+    },
+    colorStates: [{ format: 'rgba8unorm', alphaBlend: {}, colorBlend: {} }],
```
Comment (austinEng) [inline src/webgpu/api/operation/command_buffer/render/rendering.spec.ts:58]: empty object literals not needed

File: src/webgpu/api/operation/command_buffer/render/rendering.spec.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    vertexState: {
```
Comment (austinEng) [inline src/webgpu/api/operation/command_buffer/render/rendering.spec.ts:59]: vertex state not needed

File: src/webgpu/api/operation/command_buffer/render/storeop.spec.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            [[location(0)]] var<out> fragColor : vec4<f32>;
+            [[stage(fragment)]] fn main() -> void {
+              fragColor = vec4<f32>(1.0, 0.0, 0.0, 1.0);
+              return;
+            }
+            `,
+        }),
+        entryPoint: 'main',
+      },
+      layout: t.device.createPipelineLayout({ bindGroupLayouts: [] }),
```
Comment (austinEng) [inline src/webgpu/api/operation/command_buffer/render/storeop.spec.ts:53]: implicit

File: src/webgpu/api/operation/command_buffer/render/storeop.spec.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      colorAttachments: [
+        {
+          attachment: renderTexture.createView(),
+          storeOp: t.params.storeOp,
+          loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 0.0 },
+        },
+      ],
+    });
+    pass.setPipeline(renderPipeline);
+    pass.draw(3, 1, 0, 0);
```
Comment (austinEng) [inline src/webgpu/api/operation/command_buffer/render/storeop.spec.ts:70]: .draw(3)

File: src/webgpu/api/operation/command_buffer/render/rendering.spec.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            return;
+          }
+          `,
+      }),
+      entryPoint: 'main',
+    },
+    layout: pl,
+    primitiveTopology: 'triangle-list',
+    rasterizationState: {
+      frontFace: 'ccw',
```
Comment (austinEng) [inline src/webgpu/api/operation/command_buffer/render/rendering.spec.ts:56]: also probably not needed

File: src/webgpu/api/operation/command_buffer/render/rendering.spec.ts [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    colorAttachments: [
+      {
+        attachment: colorAttachmentView,
+        storeOp: 'store',
+        loadValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+      },
+    ],
+  });
+  pass.setPipeline(pipeline);
+  pass.draw(3, 1, 0, 0);
```
Comment (austinEng) [inline src/webgpu/api/operation/command_buffer/render/rendering.spec.ts:76]: .draw(3)

**Review by austinEng — APPROVED** [review]
Assuming the test is copy paste except for shaders, the shaders LGTM.

Some drive-by cleanup nits as this is an old test.

**Comment by austinEng** [comment]
Actually.. thoughts on porting to WGSL on the glsl-dependent branch and then just merging glsl-dependent into main ?
It'll make diffs easier to review, and less cross-branch merges overall.

**Comment by kainino0x** [comment]
We talked about it this morning and there's a few downsides:
- Can't easily merge back to main until the whole branch is done
- Since we roll glsl-dependent into chromium, any wgsl shaders that don't currently work for any reason would break those tests and lose coverage

You can look at just the affected file in this diff though: https://github.com/gpuweb/cts/compare/glsl-dependent..Kangz:basic_wgsl

**Review by kainino0x — APPROVED** [review]

**Comment by kainino0x** [comment]
I think all of the comments are cleanups for the old code, so I'm going to go ahead with merging this. @Kangz would you be able to do those cleanups as a followup?

**Pull request merged**